app/api/items/update_active_attempt.go

```diff
@@ -0,0 +1,58 @@
+package items
+
+import (
+	"net/http"
+
+	"github.com/go-chi/render"
+	"github.com/jinzhu/gorm"
+
+	"github.com/France-ioi/AlgoreaBackend/app/database"
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+func (srv *Service) updateActiveAttempt(w http.ResponseWriter, r *http.Request) service.APIError {
+	groupsAttemptID, err := service.ResolveURLQueryGetInt64Field(r, "groups_attempt_id")
+	if err != nil {
+		return service.ErrInvalidRequest(err)
+	}
+
+	user := srv.GetUser(r)
+	selfGroupID, err := user.SelfGroupID()
+	if err == database.ErrUserNotFound {
+		return service.InsufficientAccessRightsError
+	}
+	service.MustNotBeError(err)
+
+	var itemID int64
+	usersGroupsQuery := srv.Store.GroupGroups().WhereUserIsMember(user).Select("idGroupParent")
+	err = srv.Store.Items().Visible(user).
+		Joins("JOIN groups_attempts ON groups_attempts.idItem = items.ID AND groups_attempts.ID = ?", groupsAttemptID).
+		Joins("JOIN users_items ON users_items.idItem = items.ID AND users_items.idUser = ?", user.UserID).
+		Where("partialAccess > 0 OR fullAccess > 0").
```

Comment:
isn't it already what is done by `.Visible(.)`?

Reply:
.Visible() allows grayed access